### ISSUE TITLE
pre-trained model for Uni-20 Uni-60 baseline

### ISSUE THREAD

**Issue by bach-zouk** [riccardomarin/Diff-FMaps#3]
Hi,

Is it possible to have the pretrained model for this baseline that replicates the results in paper?

**Comment by bach-zouk** [comment]
also, I can not seem to find the data used in Figure 5 for partial shape matching in the paper.

**Comment by riccardomarin** [comment]
Hi @Not-IITian ,
Sorry for the delayed reply.

To replicate the baseline I think the easiest thing to do would be training the basis module by replacing the loss with this:

def get_uni_loss(pcA, pcB, basisA, basisB):
    # basisA_transf, C = optimal_linear_transformation(basisA, basisB)
    smax = softmax_dist(basisA, basisB)
    euc_dist = euc_dist_err(pcB, smax)
    return euc_dist

About partial shapes, the samples can be found [here](http://profs.scienze.univr.it/~marin/instant/FAUST_partials.zip).

Bests

**Comment by bach-zouk** [comment]
Hi,

Is it also possible to get the script for producing Gaussian noise in Point cloud as done for Figure 2? hopefully soon as we need to run some tests with the same noise structure for the supplementary deadline.

Thank you.

**Comment by riccardomarin** [comment]
Hi,

Sorry, I read it only now; next time, I suggest opening an issue for the specific request.

Applying Gaussian noise is more or less two lines of code; it takes less time to write it from scratch than looking for the script (that has nothing special).

Best,
Riccardo

**Comment by bach-zouk** [comment]
Thank you for telling us how many lines of code it is. we did not find any parameters of Gaussian noise in the paper or here and it is not at all obvious to even your coauthors what is difference between outlier noise and gaussian noise in terms of generating it, which is why this thread

**Comment by riccardomarin** [comment]
Something like this should do the work:
`new_vertices = vertices + 0.02 * randn(size(vertices));`

Best,
Riccardo
